app/controllers/api/v2/works_controller.rb

```diff
@@ -0,0 +1,222 @@
+class Api::V2::WorksController < Api::V2::BaseController
+  respond_to :json
+
+  # POST - search for works based on imported url
+  def search
+    works = params[:works]
+    original_urls = works.map { |w| w[:original_urls] }.flatten
+    
+    results = []
+    messages = []
+    if original_urls.nil? || original_urls.blank? || original_urls.empty?
+      status = :empty_request
+      messages << "Please provide a list of URLs to find."
+    elsif original_urls.size >= ArchiveConfig.IMPORT_MAX_CHAPTERS
+      status = :too_many_request
+      messages << "Please provide no more than #{ ArchiveConfig.IMPORT_MAX_CHAPTERS } URLs to find."
+    else
+      status = :ok
+      results = find_existing_works(original_urls)
+      messages << "Successfully searched all provided urls"
+    end
+    render_api_response(status, messages, works: results)
+  end
+
+  # POST - create a work and invite authors to claim
+  def create
+    archivist = User.find_by(login: params[:archivist])
+    external_works = params[:items] || params[:works]
+    works_responses = []
+
+    # check for top-level errors (not an archivist, no works...)
+    status, messages = batch_errors(archivist, external_works)
+
+    if status == :ok
+      # Process the works, updating the flags
+      works_responses = external_works.map { |external_work| import_work(archivist, external_work.merge(params.permit!)) }
+      success_works, error_works = works_responses.partition { |r| [:ok, :created, :found].include?(r[:status]) }
+      
+      # Send claim notification emails for successful works
+      if params[:send_claim_emails] && success_works.present?
+        notified_authors = send_external_invites(success_works, archivist)
```

Comment:
I see how this works after scrolling down to the `send_external_invites` code, but it surprised me. I didn't expect a method named `send_external_invites` to also return a list of names.